src/include/cell_inline.h

```diff
@@ -53,33 +52,82 @@ __cell_pack_value_validity(WT_SESSION_IMPL *session, uint8_t **pp, WT_TIME_WINDO
     ++*pp;
 
     flags = 0;
-    if (tw->start_ts != WT_TS_NONE) {
+    /* We pack prepared txn info to stop_ts and durable_stop_ts when:
+     *  - disagg is on
+     *  - txn is prepared
+     *  - transaction is in delete prepared (meaning it has stop_txn defined)
+     */
+    const bool pack_prepare_info_to_stop =
+      F_ISSET(S2BT(session), WT_BTREE_DISAGGREGATED) && tw->prepare && WT_TIME_WINDOW_HAS_STOP(tw);
```

Comment:
Same here as well to check for connection is preserve prepared

Reply:
yes, that's correct.

Reply:
fixed